core/src/main/java/com/graphhopper/storage/BaseGraph.java

```diff
@@ -822,41 +829,81 @@ private void setWayGeometry_( PointList pillarNodes, long edgePointer, boolean r
                 throw new IllegalArgumentException("Cannot use pointlist which is " + pillarNodes.getDimension()
                         + "D for graph which is " + nodeAccess.getDimension() + "D");
 
+            long existingGeoRef = Helper.toUnsignedLong(edges.getInt(edgePointer + E_GEO));
+            
             int len = pillarNodes.getSize();
             int dim = nodeAccess.getDimension();
-            long tmpRef = nextGeoRef(len * dim);
-            edges.setInt(edgePointer + E_GEO, Helper.toSignedInt(tmpRef));
-            long geoRef = (long) tmpRef * 4;
-            byte[] bytes = new byte[len * dim * 4 + 4];
-            ensureGeometry(geoRef, bytes.length);
-            bitUtil.fromInt(bytes, len, 0);
-            if (reverse)
-                pillarNodes.reverse();
-
-            int tmpOffset = 4;
-            boolean is3D = nodeAccess.is3D();
-            for (int i = 0; i < len; i++)
+            if (existingGeoRef > 0)
             {
-                double lat = pillarNodes.getLatitude(i);
-                bitUtil.fromInt(bytes, Helper.degreeToInt(lat), tmpOffset);
-                tmpOffset += 4;
-                bitUtil.fromInt(bytes, Helper.degreeToInt(pillarNodes.getLongitude(i)), tmpOffset);
-                tmpOffset += 4;
-
-                if (is3D)
-                {
-                    bitUtil.fromInt(bytes, Helper.eleToInt(pillarNodes.getElevation(i)), tmpOffset);
-                    tmpOffset += 4;
-                }
+            	long existingGeoRefPos = existingGeoRef * 4L;
+				final int count = wayGeometry.getInt(existingGeoRefPos);
+            	if (len <= count)
+            	{
+            		// If old array was smaller, erase contents
+            		if (len < count)
+            		{
+            			eraseWayGeometryAtGeoRef(count, existingGeoRef);
```

Comment:
I'm ok to drop the erasure. Zeroes would make it easier to see empty parts. It would also make data blocks better compressible. But that's not really necessary.

> Am 17.08.2016 um 22:34 schrieb Peter <notifications@github.com>:
> 
> In core/src/main/java/com/graphhopper/storage/BaseGraph.java:
> 
> > -                bitUtil.fromInt(bytes, Helper.degreeToInt(pillarNodes.getLongitude(i)), tmpOffset);
> > -                tmpOffset += 4;
> > -
> > -                if (is3D)
> > -                {
> > -                    bitUtil.fromInt(bytes, Helper.eleToInt(pillarNodes.getElevation(i)), tmpOffset);
> > -                    tmpOffset += 4;
> > -                }
> > +            	long existingGeoRefPos = existingGeoRef * 4L;
> > +				final int count = wayGeometry.getInt(existingGeoRefPos);
> > +            	if (len <= count)
> > +            	{
> > +            		// If old array was smaller, erase contents
> > +            		if (len < count)
> > +            		{
> > +            			eraseWayGeometryAtGeoRef(count, existingGeoRef);
> I know it is cleaner: but why would we need to clear this?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>